cellmlmanip/parser.py

```diff
@@ -432,10 +444,11 @@ def _add_connection(self, model):
                 connections_to_process.append(connection)
                 unchanged_loop_count += 1
             else:
+                connected_variable_mapping[str(connection[1])] = self.model.get_variable_by_name(connection[0])
                 unchanged_loop_count = 0
 
-            if unchanged_loop_count > len(connections_to_process):
-                raise ValueError('Unable to add connections to the model')
+            assert unchanged_loop_count <= len(connections_to_process), 'Unable to add connections to the model'
```

Comment:
I don't think this should be an assert? Looks to me it can go wrong depending on the model that's being parsed, so this is an error for the user (Exception) not for the cellmlmanip developer (assert)